server/src/group/group.service.ts

```diff
@@ -53,4 +54,47 @@ export class GroupService {
     const groupMember = await user.groupMember!.load();
     return groupMember!.group.load();
   }
+
+  /** Get group from the id.
+   * Throws a NotFoundError if the id does not correspond to a group. */
+  async getGroupFromID(id: string): Promise<Group> {
+    const group = await this.groupsRepository.findOne({ friendlyId: id });
+    if (group == null) throw NotFoundError;
+    return group;
+  }
+
+  /** Helper function that removes a user from their current group.
+   * If the user is the only member, deletes the group. */
+  async removeUserFromGroup(user: User) {
+    const oldGroup = await this.getGroupForUser(user);
+    oldGroup.members.remove(await user.groupMember!.load());
+    if (oldGroup.members.length == 0)
+      this.groupsRepository.removeAndFlush(oldGroup);
+  }
+
+  /** Adds user to an existing group, given by the group's id. */
+  async joinGroup(user: User, joinId: string) {
+    this.removeUserFromGroup(user);
+    const group = await this.getGroupFromID(joinId);
+    let groupMember: GroupMember = this.groupMembersRepository.create({
+      isHost: false,
+      user: user,
+      group: group,
+    });
+
+    group.members.set([groupMember]);
+    user.groupMember?.set(groupMember);
+    groupMember.user.set(user);
+
+    await this.groupsRepository.persistAndFlush(group);
+  }
+
+  /** Moves the user out of their current group into a new group */
+  async leaveGroup(user: User) {
```

Comment:
Same thing as in removeUserFromGroup, but you'll have to do this some other way than joinGroup since you do not use removeUserFromGroup.